### ISSUE TITLE
Fix flaky Angular tests

### ISSUE THREAD

**Issue by amdw** [google/peoplemath#72]
After setting up the continuous build, I noticed a few unrepeatable failures in the Angular tests. In a typical failed build, a single test would fail, with a stack trace like this:

```
Chrome Headless 85.0.4183.102 (Linux x86_64) ERROR
  An error was thrown in afterAll
  TypeError: this.storage.updatePeriod is not a function
      at <Jasmine>
      at PeriodComponent.performSave (http://localhost:9876/_karma_webpack_/src/app/period/period.component.ts:333:18)
      at SafeSubscriber._next (http://localhost:9876/_karma_webpack_/src/app/period/period.component.ts:67:75)
      at SafeSubscriber.__tryOrUnsub (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm2015/internal/Subscriber.js:183:1)
      at SafeSubscriber.next (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm2015/internal/Subscriber.js:122:1)
      at Subscriber._next (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm2015/internal/Subscriber.js:72:1)
      at Subscriber.next (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm2015/internal/Subscriber.js:49:1)
      at DebounceTimeSubscriber.debouncedNext (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm2015/internal/operators/debounceTime.js:40:1)
      at AsyncAction.dispatchNext (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm2015/internal/operators/debounceTime.js:53:1)
      at AsyncAction._execute (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm2015/internal/scheduler/AsyncAction.js:51:1)
      at AsyncAction.execute (http://localhost:9876/_karma_webpack_/node_modules/rxjs/_esm2015/internal/scheduler/AsyncAction.js:39:1)
```

I'm not sure exactly what's going on here, but this feels like it might be a race condition of some kind.

**Comment by amdw** [comment]
These tests are definitely flaky in the build. The Cloud Build seems to fail approximately 15% of the time with an error of this type.

**Comment by amdw** [comment]
I hope I'm not speaking too soon, but I think I know why this is happening. If I'm right, the problem is this test:

https://github.com/google/peoplemath/blob/6ee6b31205474e48b037a2f550563a2fb48b7c27/src/app/period/period.component.spec.ts#L92

That test indirectly calls `save()` on the period, which calls `updatePeriod()` on the `storage` asynchronously using RxJS `debounceTime`. However, the test is not `waitForAsync`, so it does not wait for this callback to fire.

When you wrap that test in `waitForAsync`, it consistently fails, with a stack trace exactly like the one above. The problem is that `updatePeriod` is not mocked out on the Jasmine spy object for the `storage`.

The callback fires 5000ms later. My theory is that if the tests have all finished within those 5 seconds, everything works fine, but if they haven't, the callback throws an exception, and that exception gets (very confusingly) associated with whichever test happens to be running at that point.

Proposed steps:

* Use `fakeAsync` for this test (since there's no reason to wait two real seconds for the timer to fire)
* Fix the spy to stub out `updatePeriod()` (to fix the test)
* Wrap lots of other tests in `waitForAsync`. I see no harm in doing this, just to be sure that if we ever forget and trigger an asynchronous save in any of them, then any failures will be deterministic and clearly associated with the test that is the culprit.

**Comment by amdw** [comment]
I believe this is fixed in #100. I'll reopen this if this proves not to be true.
